brushfire-features/src/main/scala/com/stripe/brushfire/features/DispatchedFeatureEncoder.scala

```diff
@@ -0,0 +1,232 @@
+
+import com.twitter.algebird.{ Aggregator, Semigroup, Monoid }
+
+case class DispatchedFeatureEncoding[L](encoder: DispatchedFeatureEncoder)
+extends FeatureEncoding[String, FeatureValue, Map[L, Long]] with Defaults {
+  val splitter: Splitter[FeatureValue, Map[L, Long]] =
+    DispatchedSplitter(
+      BinarySplitter[Double, Map[L, Long]](LessThan(_)),
+      BinarySplitter[String, Map[L, Long]](EqualTo(_)),
+      doubleSplitter[Map[L, Long]],
+      SpaceSaverSplitter[String, L]()
+    )
+}
+
+object DispatchedFeatureEncoding {
+  import DispatchedFeatureEncoder.Counts
+
+  def trainer[L](
+    platform: TrainingPlatform
+  )(implicit
+    ev: platform.AggregateContext[Map[String, Option[Counts]]]
+  ): platform.Trainer[Map[String, FValue], DispatchedFeatureEncoding[L]] = {
+    platform.Trainer.aggregate(DispatchedFeatureEncoder.aggregator).map(DispatchedFeatureEncoding[L](_))
+  }
+}
+
+sealed trait DispatchedFeatureType
+object DispatchedFeatureType {
+  case object Ordinal extends DispatchedFeatureType
+  case object Nominal extends DispatchedFeatureType
+  case object Continuous extends DispatchedFeatureType
+  case object Sparse extends DispatchedFeatureType
+  case class BooleanCategory(tval: String, fval: String) extends DispatchedFeatureType
+}
+
+case class DispatchedFeatureEncoder(
+  mappings: Map[String, DispatchedFeatureType]
+) extends FeatureEncoder[String, FeatureValue] {
+  def encode(value: Map[String, FValue]): Try[Map[String, FeatureValue]] = Try {
+    val bldr = Map.newBuilder[String, FeatureValue]
+
+    value.foreach { case (k, v) =>
+      mappings.get(k).foreach { tpe =>
+        val dispatched = tpe match {
+          case DispatchedFeatureType.Ordinal =>
+            Ordinal(v.tryAsDouble.get)
+          case DispatchedFeatureType.Nominal =>
+            Nominal(v.tryAsString.get)
+          case DispatchedFeatureType.Continuous =>
+            Continuous(v.tryAsDouble.get)
+          case DispatchedFeatureType.Sparse =>
+            Sparse(v.tryAsString.get)
+          case DispatchedFeatureType.BooleanCategory(t, f) =>
+            if (v.tryAsBoolean(t, f).get) {
+              Nominal(t)
+            } else {
+              Nominal(f)
+            }
+        }
+
+        bldr += (k -> dispatched)
+      }
+    }
+
+    bldr.result()
+  }
+}
+
+object DispatchedFeatureEncoder {
+  val aggregator: Aggregator[Map[String, FValue], Map[String, Option[Counts]], DispatchedFeatureEncoder] =
+    MapAggregator[String, FValue, Option[Counts], Option[DispatchedFeatureType]](Counts.aggregator)
+      .andThenPresent { ts =>
+        DispatchedFeatureEncoder(ts.collect { case (key, Some(value)) => key -> value })
+      }
+
+  def isDouble(s: String): Boolean = Try(s.toDouble).isSuccess
+
+  case class UniqueCount[A](uniques: Option[Set[A]]) {
+    def ++ (that: UniqueCount[A]): UniqueCount[A] =
+      UniqueCount(for {
+        a <- this.uniques
+        b <- that.uniques
+        c = a ++ b
+        if (c.size < 20)
```

Comment:
As @avibryant mentioned earlier, this should probably be bumped up or made configurable.